src/vs/editor/common/viewModel/viewModelDecorations.ts

```diff
@@ -115,6 +121,48 @@ export class ViewModelDecorations implements IDisposable {
 		return this._getDecorationsInRange(range, onlyMinimapDecorations, onlyMarginDecorations).inlineDecorations[0];
 	}
 
+	/**
+	 * Get an array where each index maps the line number to its line height.
+	 *
+	 * Index 0 is not an actual line number. It always contains the original line height.
+	 */
+	public getDecorationsLineHeightMap(): Uint8ClampedArray {
+		if (!this._decorationsHeightMapCache) {
+			const lineCount = this._linesCollection.getViewLineCount();
+			const lineHeights = new Uint8ClampedArray(lineCount + 1).fill(this._lineHeight);
+			const viewRange = new Range(1, this._linesCollection.getViewLineMinColumn(0), lineCount, this._linesCollection.getViewLineMaxColumn(lineCount));
+			const modelDecorations = this._linesCollection.getDecorationsInRange(viewRange, this.editorId, true, false, false);
+
+			for (const decoration of modelDecorations) {
+				const decorationLineHeight = decoration.options.lineHeight;
+
+				if (!decorationLineHeight) {
+					continue;
+				}
+
+				const range = this._linesCollection.convertModelRangeToViewRange(decoration.range);
+				for (let rangeLine = range.startLineNumber; rangeLine <= range.endLineNumber; rangeLine++) {
+					lineHeights[rangeLine] = Math.max(lineHeights[rangeLine], decorationLineHeight);
+				}
+			}
+
+			this._decorationsHeightMapCache = lineHeights;
+		}
+
+		return this._decorationsHeightMapCache;
+	}
+
+	public getDecorationsOffset(lineNumber: number = this._linesCollection.getViewLineCount()): number {
```

Comment:
`lineNumber: number = this._linesCollection.getViewLineCount()`

I think this should be `= this._linesCollection.getViewLineCount() + 1`.